fedimint-api-client/src/api.rs

```diff
@@ -1257,6 +1316,13 @@ impl WsFederationApi<WsClient> {
         )
     }
 
+    pub fn with_self_peer_id(self, self_peer_id: PeerId) -> Self {
+        Self {
+            self_peer_id: Some(self_peer_id),
+            ..self
+        }
+    }
+
     /// Creates a new API client from a invite code, assumes they are in peer
     /// id order
     pub fn from_invite_code(info: &[InviteCode]) -> Self {
```

Comment:
unrelated to this PR: this fn is such a foot-gun. Could be written much more elegantly now that single invite codes can contain multiple peer URLs.